grudge/dof_desc.py

```diff
@@ -208,29 +265,43 @@ def is_trace(self):
                     FACE_RESTR_INTERIOR])
 
     def uses_quadrature(self):
-        if self.quadrature_tag is None:
-            return False
-        if self.quadrature_tag is QTAG_NONE:
-            return False
-
-        return True
-
-    def with_qtag(self, qtag):
-        return type(self)(domain_tag=self.domain_tag, quadrature_tag=qtag)
+        # FIXME: String tags are deprecated
+        # Check for string first, otherwise
+        # `issubclass` will raise an exception whenever
+        # its first argument is not a class.
+        # This can go away once support for strings is dropped
+        # completely.
+        if isinstance(self.discretization_tag, str):
+            return True
+        if issubclass(self.discretization_tag, DISCR_TAG_QUAD):
+            return True
+
+        return False
```

Comment:
I'm not sure I'm loving this fallback to `False` here. I'd prefer a structure where the `else` branch is "I don't know what I'm looking at, help?" (i.e. an error)

Reply:
Good call! Done in https://github.com/inducer/grudge/pull/83/commits/d778430473a03d93f750e6010a614e9c140d94f7